core/mutator/entry/mutator.go

```diff
@@ -87,6 +87,10 @@ func ReduceFn(leaves []*pb.EntryUpdate, msgs []*pb.EntryUpdate,
 	// Filter for mutations that are valid.
 	newEntries := make([]*pb.EntryUpdate, 0, len(msgs))
 	for i, msg := range msgs {
+		// Skip if there is no change in data.
+		if len(leaves) > 0 && bytes.Equal(leaves[0].GetCommitted().GetData(), msg.GetCommitted().GetData()) {
```

Comment:
For more correct code, do the `if len(leaves) > 0` check separately and return an error if it is false. 
This is already done for you on line 78. Consider setting a new oldCommitment value there. 

What we're doing here is defining equality for two EntryUpdates.  Perhaps we should create an explicit `Equals` method for them and check
- The commitment - as we are doing here
- AuthorizedKeyset
- Index

Reply:
I added the check for AuthorizedKeyset. The "index" should be equal already since the "reduce" is based on the index.